Index: SDL-1.2.15/src/main/macosx/SDLMain.m
===================================================================
--- SDL-1.2.15/src/main/macosx/SDLMain.m	(revision 1000)
+++ SDL-1.2.15/src/main/macosx/SDLMain.m	(revision 996)
@@ -20,6 +20,22 @@
 /* Use this flag to determine whether we use SDLMain.nib or not */
 #define		SDL_USE_NIB_FILE	0
 
+/* Use this flag to determine whether we use CPS (docking) or not */
+#define		SDL_USE_CPS		1
+#ifdef SDL_USE_CPS
+/* Portions of CPS.h */
+typedef struct CPSProcessSerNum
+{
+	UInt32		lo;
+	UInt32		hi;
+} CPSProcessSerNum;
+
+extern OSErr	CPSGetCurrentProcess( CPSProcessSerNum *psn);
+extern OSErr 	CPSEnableForegroundOperation( CPSProcessSerNum *psn, UInt32 _arg2, UInt32 _arg3, UInt32 _arg4, UInt32 _arg5);
+extern OSErr	CPSSetFrontProcess( CPSProcessSerNum *psn);
+
+#endif /* SDL_USE_CPS */
+
 static int    gArgc;
 static char  **gArgv;
 static BOOL   gFinderLaunch;
@@ -185,7 +201,18 @@
     SDLMain				*sdlMain;
 
     /* Ensure the application object is initialised */
-    [[NSApplication sharedApplication] activateIgnoringOtherApps: YES];
+    [NSApplication sharedApplication];
+    
+#ifdef SDL_USE_CPS
+    {
+        CPSProcessSerNum PSN;
+        /* Tell the dock about us */
+        if (!CPSGetCurrentProcess(&PSN))
+            if (!CPSEnableForegroundOperation(&PSN,0x03,0x3C,0x2C,0x1103))
+                if (!CPSSetFrontProcess(&PSN))
+                    [NSApplication sharedApplication];
+    }
+#endif /* SDL_USE_CPS */
 
     /* Set up the menubar */
     [NSApp setMainMenu:[[NSMenu alloc] init]];
